src/api/providers/nebius.ts

```diff
@@ -0,0 +1,87 @@
+import { Anthropic } from "@anthropic-ai/sdk"
+import OpenAI from "openai"
+import { withRetry } from "../retry"
+import { ApiHandler } from "../index"
+import { convertToOpenAiMessages } from "../transform/openai-format"
+import { ApiStream } from "../transform/stream"
+import { convertToR1Format } from "../transform/r1-format"
+import {
+	nebiusDefaultURL,
+	nebiusDefaultModelId,
+	nebiusModels,
+	ModelInfo,
+	ApiHandlerOptions,
+	nebiusModelId,
+} from "../../shared/api"
+
+export class NebiusHandler implements ApiHandler {
+	private options: ApiHandlerOptions
+	private client: OpenAI
+
+	constructor(options: ApiHandlerOptions) {
+		this.options = options
+		this.client = new OpenAI({
+			baseURL: nebiusDefaultURL,
+			apiKey: this.options.nebiusApiKey,
+		})
+	}
+
+	@withRetry()
+	async *createMessage(systemPrompt: string, messages: Anthropic.Messages.MessageParam[]): ApiStream {
+		const modelId = this.options.nebiusModelId ?? ""
+		const isDeepseekReasoner = modelId.includes("deepseek-reasoner")
+
+		let openAiMessages: OpenAI.Chat.ChatCompletionMessageParam[] = [
+			{ role: "system", content: systemPrompt },
+			...convertToOpenAiMessages(messages),
+		]
+
+		if (isDeepseekReasoner) {
+			openAiMessages = convertToR1Format([{ role: "user", content: systemPrompt }, ...messages])
+		}
+
+		const stream = await this.client.chat.completions.create({
+			model: modelId,
+			messages: openAiMessages,
+			temperature: 0,
+			stream: true,
+			stream_options: { include_usage: true },
+		})
+		for await (const chunk of stream) {
+			const delta = chunk.choices[0]?.delta
+			if (delta?.content) {
+				yield {
+					type: "text",
+					text: delta.content,
+				}
+			}
+
+			if (delta && "reasoning_content" in delta && delta.reasoning_content) {
+				yield {
+					type: "reasoning",
+					reasoning: (delta.reasoning_content as string | undefined) || "",
+				}
+			}
+
+			if (chunk.usage) {
+				yield {
+					type: "usage",
+					inputTokens: chunk.usage.prompt_tokens || 0,
+					outputTokens: chunk.usage.completion_tokens || 0,
+				}
+			}
+		}
+	}
+
+	getModel(): { id: nebiusModelId; info: ModelInfo } {
+		const modelId = this.options.nebiusModelId ?? ""
+		if (modelId && modelId in nebiusModels) {
+			const id = modelId as nebiusModelId
+			return { id, info: nebiusModels[id] }
+		}
+		return {
```

Comment:
NOTE: The model used here will default to model listed in this constant 

<img width="588" alt="image" src="https://github.com/user-attachments/assets/83cad904-5e14-4ac3-889f-d9e674fd3c94" />


Unless the model provided by the user is one of the models in this string. So what this means is that the users can't use all the models but only the ones in this list correct?

Reply:
@kitcat-dev Thanks for the PR, it works as expected a minor UX recommendation.

Try selecting 
<img width="423" alt="image" src="https://github.com/user-attachments/assets/768a12d4-b20f-42f8-9fb4-c341e5eb4374" />

The qwen models while scrolling in the UI and you might realize the UX for that is not the cleanest and you might need to use downshift keys for it. Feel free to make a follow up PR for that.